codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2HeaderTableListSizeTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.http2;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests for {@link DefaultHttp2HeaderTableListSize}.
+ */
+public class DefaultHttp2HeaderTableListSizeTest {
+
+    private DefaultHttp2HeaderTableListSize headerTable;
+
+    @Before
+    public void setup() {
+        headerTable = new DefaultHttp2HeaderTableListSize();
+    }
+
+    @Test
+    public void defaultMaxHeaderListSizeShouldSucceed() {
+        assertEquals(Integer.MAX_VALUE, (long) headerTable.maxHeaderListSize());
+    }
+
+    @Test
+    public void standardMaxHeaderListSizeShouldSucceed() {
+        try {
+            headerTable.maxHeaderListSize(123);
+            assertEquals(123L, (long) headerTable.maxHeaderListSize());
+        } catch (Http2Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void boundaryMaxHeaderListSizeShouldSucceed() {
+        try {
+            headerTable.maxHeaderListSize(Integer.MAX_VALUE);
+            assertEquals(Integer.MAX_VALUE, (long) headerTable.maxHeaderListSize());
+        } catch (Http2Exception e) {
+            fail();
+        }
+
+        try {
+            final long settingsValueUpperBound = (1L << 32) - 1L;
+            headerTable.maxHeaderListSize((int) settingsValueUpperBound);
+            assertEquals(Integer.MAX_VALUE, (long) headerTable.maxHeaderListSize());
+        } catch (Http2Exception e) {
+            fail();
+        }
+    }
```

Comment:
Should you also have a test with setting something like `Long.MAX_VALUE` and verify that the value set was `Integer.MAX_VALUE`?